chapter_explainability/global_model_analysis.md

```diff
@@ -0,0 +1,265 @@
+
+The following line can diagnose the error distribution across cohorts of different loan terms.
+
+```{.python .input  n=9}
+explainer.draw_heat_map(error.term, error.Status)
+```
+
+From this error map, we observe the difference of error rate on different loan term (36 months vs. 60 months). Evidently, the model makes less mistakes on long-term loans. If the feature of interest is sensitive (e.g., gender), this kind of disparity will lead to unacceptable discrimination and unfaireness. Error stratification can keep us aware of possible model biases.
+
+Similarly, we can also stratify the erroneous cases based on the annual income by classifying the income into different levels.
+
+```{.python .input  n=10}
+explainer.draw_heat_map(pd.qcut(error.annual_inc, q=3), error.Status)
+```
+
+We can even display the error distributions at a finer granularity with feature crossing. In this example, we consider two features: the term of loan and whether the borrower fully owns their house (OWN: 1=Yes, 0=No).
+
+```{.python .input  n=11}
+explainer.draw_heat_map([error.term, error.OWN], error.Status)
+```
+
+The error analyses mentioned above are conducted automatically, but it is also viable to stratify the error cases manually. For example, when analyzing an image classifier, we can manually define the cohorts based on images' patterns (e.g., whether an image is blurry). 
+
+
+## Ablation Study
+
+Ablation studies refer to removing certain features (or components) of the model and seeing the features' (or components') contribution to the overall model performance. Some models, such as random forests provide an innate way of calculating feature importance (e.g., using Gini or entropy) :cite:`Breiman.2001`, but these methods are model dependent. Ablation studies are model-agnostic and can be applied to any kind of model.
+
+
+### Global Feature Importance via Ablation Study
+Feature importance refers to techniques that assign a score to input features based on how useful they are at predicting a target variable. With a dataset of $m$ features, the procedure goes like this:
+
+* Train the model on the entire training set and obtain an accuracy score $s$ (or other scoring metric) on the validation set.
+* For each feature $i$, $i \in m$, remove it from the training set, then retrain the model and get another score $s_i$ on the validation set. Alternatively, we can permute the feature and then re-evaluate the model :cite:`Breiman.2001`. Feature permutation is more efficient as it does not require model retraining.
+* Calculate the global feature importance score with $s-s_i$.
+
+Now, let's implement this feature ablation method. The following function provides two ablation modes: *REMOVAL* and *SHUFFLE*. To permute the feature $i$, we randomly shuffle the values of feature $i$ in the current batch.
+
+```{.python .input  n=12}
+@d2l.add_to_class(GlobalExplainer)
+def feature_importance(self, feat_of_interest, mode="SHUFFLE"):
```

Comment:
If `feat_of_interest` is a list, plural naming enhances readability: `features_of_interest`